units/concepts.py

```diff
@@ -0,0 +1,112 @@
+from functools import partial
+from itertools import groupby
+
+import structlog
+import httpx
+
+from units.settings import get_settings
+
+logger = structlog.get_logger()
+
+
+def remove_graph_namespaces(s: str) -> str:
+    prefixes = (
+        "http://qudt.org/schema/qudt/",
+        "http://www.w3.org/1999/02/22-rdf-syntax-ns#",
+        "http://www.w3.org/2004/02/skos/core#",
+    )
+
+    for prefix in prefixes:
+        if s.startswith(prefix):
+            return s.replace(prefix, "")
+    return s
+
+
+def language_filter(o: dict, lang: str) -> dict:
+    """Check if given element has the correct language, if one is given"""
+    if "xml:lang" in o and lang not in o["xml:lang"].lower():
+        return False
+    return True
+
+
+def get_qk_for_iri(iri: str) -> str | None:
+    """Get the QUDT quantity key for a given unit IRI."""
+    settings = get_settings()
+    logger.debug("Using sparql endpoint url %s", settings.UNITS_SPARQL_URL)
+
+    QUERY = f"""
+PREFIX qudt: <http://qudt.org/schema/qudt/>
+
+JSON {{
+    "this is ignored anyway": ?qk
+}}
+FROM <{settings.UNITS_VOCAB_PREFIX}qudt/>
+FROM <{settings.UNITS_VOCAB_PREFIX}simapro/>
```

Comment:
Is this going to expand over time? I guess it could later be dynamic?

Reply:
Rather the opposite - see [comment ](https://github.com/Depart-de-Sentier/brightcon-2024-material/issues/1#issuecomment-2370351664)